### PR TITLE
Makefile,Vagrantfile: use box v0.10.2 & Go 1.7.6

### LINKED ISSUE #875: go 1.6.2 does not compile code.
Need to bump up minimum version of go in the makefile.

### PR DESCRIPTION
**Pull request by unclejack**
##### Closes #875
This PR bumps the CentOS Vagrant box we're using to 0.10.2. This new box bundles Go 1.7.6 which fixes one security issue.

This new version of the CentOS Vagrant box includes a newer kernel. etcd and the other components are as they were.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -18,7 +18,7 @@ TAR_EXT := tar.bz2
 TAR_FILENAME := $(NAME)-$(VERSION).$(TAR_EXT)
 TAR_LOC := .
 TAR_FILE := $(TAR_LOC)/$(TAR_FILENAME)
-GO_MIN_VERSION := 1.5.1
+GO_MIN_VERSION := 1.7
```
Comment (dseevr) [inline Makefile]: Technically, this addresses https://github.com/contiv/netplugin/issues/875

Comment (unclejack) [inline Makefile]: I've added "Closes #875" to the main comment to close that issue when this gets merged.

**Comment by rhim** [comment]
I remember other issues with Go1.7? Are those addressed?

**Comment by unclejack** [comment]
@rhim: The issues which we were discussing were related to Go 1.8. Fortunately, we've been using Go 1.7.x to compile netplugin for a while without running into issues.

**Comment by rhim** [comment]
Can we consolidate the build to a container to capture the dependencies at a single place?

**Comment by unclejack** [comment]
@rhim: More work is needed to be able to do that. It's out of the scope of this PR.

**Pull request merged**